tests/document/indexes.py

```diff
@@ -911,22 +910,26 @@ class BlogPost(Document):
             meta = {'db_alias': 'test_indexes_after_database_drop'}
 
         try:
+            # Create Post #1
+            BlogPost(title='test1', slug='test').save()
+            # Re-create Post #1
+            with self.assertRaises(NotUniqueError):
+                BlogPost(title='test1', slug='test').save()
+
             BlogPost.drop_collection()
 
-            # Create Post #1
-            post1 = BlogPost(title='test1', slug='test')
-            post1.save()
+            # Create Post #1 again
+            BlogPost(title='test1', slug='test').save()
 
             # Drop the Database
             connection.drop_database('tempdatabase')
```

Comment:
@wojcikstefan 

> What happens if you initialize a document, ensure indexes, then drop the collection, and then ensure indexes again?

You are correct. That was a bad idea. There is already a test for that. I refactored the changes removing creation of index each time in `save` and this test fails after `connection.drop_database('tempdatabase')`. We it would be possible to hook into this, I would clear `_collection` attribute on all models. I don't have a solution for this. :(

The only way out is not use index auto-creation.